dev/InfoBar/TestUI/Themes/InfoBarStyles.xaml

```diff
@@ -0,0 +1,156 @@
+                                            <Setter.Value>
+                                                <FontIconSource FontFamily="Segoe MDL2 Assets" Glyph="&#xEA39;"/>
+                                            </Setter.Value>
+                                        </Setter>
+                                    </VisualState.Setters>
+                                </VisualState>
+                                <VisualState x:Name="Warning">
+                                    <VisualState.Setters>
+                                        <Setter Target="Brush.Color" Value="#FF8C00"/>
+                                        <Setter Target="Icon.IconSource">
+                                            <Setter.Value>
+                                                <FontIconSource FontFamily="Segoe MDL2 Assets" Glyph="&#xE783;"/>
+                                            </Setter.Value>
+                                        </Setter>
+                                    </VisualState.Setters>
+                                </VisualState>
+                                <VisualState x:Name="Informational">
+                                    <VisualState.Setters>
+                                        <Setter Target="Brush.Color" Value="#0078D7"/>
+                                        <Setter Target="Icon.IconSource">
+                                            <Setter.Value>
+                                                <FontIconSource FontFamily="Segoe MDL2 Assets" Glyph="&#xE946;"/>
+                                            </Setter.Value>
+                                        </Setter>
+                                    </VisualState.Setters>
+                                </VisualState>
+                                <VisualState x:Name="Success">
+                                    <VisualState.Setters>
+                                        <Setter Target="Brush.Color" Value="#107C10"/>
+                                        <Setter Target="Icon.IconSource">
+                                            <Setter.Value>
+                                                <FontIconSource FontFamily="Segoe MDL2 Assets" Glyph="&#xF13E;"/>
+                                            </Setter.Value>
+                                        </Setter>
+                                    </VisualState.Setters>
+                                </VisualState>
+                                <VisualState x:Name="None">
+                                    <VisualState.Setters>
+                                        <Setter Target="Brush.Color" Value="Transparent"/>
+                                    </VisualState.Setters>
+                                </VisualState>
+                                <VisualState x:Name="Default"/>
+                            </VisualStateGroup>
+                            <VisualStateGroup x:Name="UserCustomStates">
```

Comment:
This is an interesting approach, but I don't think it is going to work well. It implies that a custom IconSource and a custom StatusColor are mutually exclusive, which is not correct.

Also, since both this group and the SeverityLevels group set the Icon.IconSource and Brush.Color properties, it depends on the order of the GoToState calls. So if the SeverityLevel changes, it will run again and will overwrite the UserCustomStates value.

I think the best approach here is to use a TemplateSettings object to hold the values of the brushes. Or you could have two rectangles - a usercolor rectangle and a standardcolor rectangle. Only one rectangle is visible at a time - if InfoBar.StatusColor is set then usercolor rectangle is Visible and standardcolor rectangle is Collapsed. If InfoBar.StatusColor is unset. the standardColor rectangle is Visible and usercolor rectangle is Collapsed (you would have a VisualStateGroup to control this). SeverityLevels VisualStateGroup changes the color of standardcolor rectangle - which is only visible if InfoBar.StatusColor is not set. That approach might work.

Reply:
> Also, since both this group and the SeverityLevels group set the Icon.IconSource and Brush.Color properties, it depends on the order of the GoToState calls. So if the SeverityLevel changes, it will run again and will overwrite the UserCustomStates value.

Yeah, I see what you mean! The way it works now is that the user can override the severity level's StatusColor and Icon when they first define the InfoBar in xaml (UserCustomState visual state manager is only accessed in OnApplyTemplate). Then, when the app is running, if you wanted to choose a different severity level, it would take on that severity level's color and icon (or unless the user wants to set it to another specific statuscolor or icon). Hm, I thought it could be a desirable feature so the user could have more freedom. I guess this kind of goes back to the discussion from above. Thoughts? @gabbybilka

Reply:
Okay. With the most recent changes, I have included two rectangles, as you've described. I've also included two icons (StandardIcon and UserIcon). Now, if the user has set their own custom color/icon, even if they change the severity level, the custom color/icon will remain.